src/metabase/upload.clj

```diff
@@ -339,6 +340,25 @@
          :num-rows          (count (rest rows))
          :generated-columns 0}))))
 
+(defn- create-from-csv-and-sync!
+  "This is separated from `create-csv-upload!` for testing"
+  [{:keys [db file schema table-name]}]
+  (let [driver            (driver.u/database->driver db)
+        schema            (some->> schema (ddl.i/format-name driver))
+        table-name        (some->> table-name (ddl.i/format-name driver))
```

Comment:
We now need to format the schema and table names by driver, because Redshift only returns lower case names from system tables for sync even if the table is created with upper case letters. Without this chnage, sync wouldn't pick up the table because we are doing case-sensitive matching there